examples/discrete_turbines/channel-optimisation.py

```diff
@@ -166,7 +166,8 @@ def eval_cb_pre(controls):
 
 def derivative_cb_pre(controls):
     print_output("ADJOINT RUN:")
-    print_output("positions: {}".format([float(c_) for c_ in controls]))
+    print_output("positions: {}".format([float(c_.data()) for c_ in controls]))
```

Comment:
Out of interest, why was this change needed? Does taking the `float` of a `Constant` no longer do what you expect?

Reply:
Ah, because if you create a `Constant(..., domain=...)` you no longer actually get an object of type `Constant` it actually returns a `Function` on the R FunctionSpace. And taking the float of a Function is not defined of course.

Reply:
Ah okay makes sense, thanks.